apsuite/commissioning_scripts/meas_coupling_tune.py

```diff
@@ -17,201 +41,203 @@ class FitTunes(_SimulAnneal):
-        """."""
-        self.position[4] = value
-
-    def calc_obj_fun(self):
-        """."""
-        # print(self.position)
-        coeff1, offset1, coeff2, offset2, coupling = self.position
-        _, tune1, tune2 = \
-            FitTunes.calc_tunes(
-                self._param, coeff1, offset1, coeff2, offset2, coupling)
-        diff = ((tune1 - self._tune1)**2 + (tune2 - self._tune2)**2)**0.5
-        residue = _np.mean(diff)
-        # print(self.position, residue)
-        return residue
-
-    def initialization(self):
-        """."""
-        self._calc_init_parms()
-        coeff1, offset1, coeff2, offset2, _ = self.position
-        self.deltas = _np.array([
-            0.01 * 0.002 * (coeff1 - coeff2),
-            0.01 * 0.002 * (offset1 - offset2),
-            0.01 * 0.002 * (coeff1 - coeff2),
-            0.01 * 0.002 * (offset1 - offset2),
-            0.5 * 0.1/100])
-
-    @staticmethod
-    def calc_tunes(
-            param=None,
-            coeff1=None, offset1=None,
-            coeff2=None, offset2=None, coupling=None):
-        """."""
-        fx_ = coeff1 * param + offset1
-        fy_ = coeff2 * param + offset2
-        tune1, tune2 = _np.zeros(param.shape), _np.zeros(param.shape)
-        for i in range(len(param)):
-            mat = _np.array([[fx_[i], coupling/2], [coupling/2, fy_[i]]])
-            eigvalues, _ = _np.linalg.eig(mat)
-            tune1[i], tune2[i] = eigvalues
-            if tune1[i] <= tune2[i]:
-                tune1[i], tune2[i] = tune2[i], tune1[i]
-        return param, tune1, tune2
-
-    def fitting_plot(
-            self, oversampling=1, title=None, xlabel=None, fig=None):
-        """."""
-        position = self.position
-        param = _np.interp(
-            _np.arange(0, len(self._param), 1/oversampling),
-            _np.arange(0, len(self._param)),
-            self._param)
-
-        # NOTE: do error estimate properly!
-        coup_error = self.calc_obj_fun()
+    def ismeasuring(self):
+        """."""
+        return self._thread.is_alive()
+
+    def _do_meas(self):
+        quad = self.devices['quad']
+        tunes = self.devices['tune']
+
+        curr0 = quad.current
+        curr_vec = curr0 * _np.linspace(
+            1-self.params.neg_percent,
+            1+self.params.pos_percent,
+            self.params.nr_points)
+        tunes_vec = _np.zeros((len(curr_vec), 2))
+
+        print('QFB Current:')
+        for idx, curr in enumerate(curr_vec):
+            quad.current = curr
+            _time.sleep(self.params.time_wait)
+            tunes_vec[idx, :] = tunes.tunex, tunes.tuney
+            print('   {:.6f} A ({:+.3f} %), nux = {:.4f}, nuy = {:.4f}'.format(
+                curr, (curr/curr0-1)*100),
+                tunes_vec[idx, 0], tunes_vec[idx, 1])
+        print('Finished!')
+        quad.current = curr0
+        self.data['qname'] = quad.devname
+        self.data['current'] = curr_vec
+        self.data['tunes'] = tunes_vec
```

Comment:
I think it is important to save the time when the measurement was performed, so we can recover additional information via archiver later on.